lib/active_record_shards/connection_switcher.rb

```diff
@@ -162,11 +167,6 @@ def shard_names
 
     def switch_connection(options)
       if options.any?
-        if options.key?(:slave)
-          ActiveRecordShards::Deprecation.warn('the `:slave` option should be replaced with `:replica`!')
-          options[:replica] ||= options.delete(:slave)
-        end
```

Comment:
why is this being removed?

Reply:
this is removed because we're changing options in ShardSelection, which is what gets passed in here:

```
def options
      { shard: @shard, replica: @on_replica } # no more :slave option
end
```

Reply:
I suppose if we proceed with removing it below, then this makes sense.